rts/at/tx/src/main/java/org/jboss/jbossts/star/resource/Transaction.java

```diff
@@ -323,7 +323,9 @@ protected boolean afterCompletion(int arjunaStatus, boolean report_heuristics) {
             }
         }
 
-        coordinator.removeTxState(arjunaStatus, this, enlistmentIds);
+        if (failedList == null || failedList.size() == 0) {
```

Comment:
Wait, I already have a tests for that:
https://github.com/jbosstm/narayana/blob/master/rts/at/tx/src/test/java/org/jboss/jbossts/star/test/SpecTest.java#L426
and
https://github.com/jbosstm/narayana/blob/master/rts/at/tx/src/test/java/org/jboss/jbossts/star/test/SpecTest.java#L371